apps/wallet-mobile/src/features/AddWallet/useCases/CreateWallet/RecoveryPhrase.tsx

```diff
@@ -0,0 +1,204 @@
+  ScrollView,
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+  useWindowDimensions,
+  View,
+} from 'react-native'
+import LinearGradient from 'react-native-linear-gradient'
+import {SafeAreaView} from 'react-native-safe-area-context'
+
+import {Button, StatusBar, useModal} from '../../../../components'
+import {Space} from '../../../../components/Space/Space'
+import {CardAboutPhrase} from '../../common/CardAboutPhrase/CardAboutPhrase'
+import {LearnMoreButton} from '../../common/LearnMoreButton/LearnMoreButton'
+import {mockAddWallet} from '../../common/mocks'
+import {StepperProgress} from '../../common/StepperProgress/StepperProgress'
+import {useStrings} from '../../common/useStrings'
+import EyeClosedIcon from '../../illustrations/EyeClosed'
+import EyeOpenIcon from '../../illustrations/EyeOpen'
+import InfoIcon from '../../illustrations/InfoIcon'
+
+export const RecoveryPhrase = () => {
+  const {styles, colors} = useStyles()
+  const {openModal, closeModal} = useModal()
+  const [isBlur, setIsBlur] = React.useState(true)
+
+  const navigation = useNavigation()
+
+  const HEIGHT_SCREEN = useWindowDimensions().height
+  const PERCENTAGE = HEIGHT_SCREEN >= 900 ? 48 : HEIGHT_SCREEN >= 800 ? 55 : 75
+  const HEIGHT_MODAL = (HEIGHT_SCREEN / 100) * PERCENTAGE
+
+  const strings = useStrings()
+
+  const showModal = () => {
+    Keyboard.dismiss()
+    openModal(
+      strings.recoveryPhraseModalTitle,
+      <View style={styles.modal}>
+        <ScrollView>
+          <View style={styles.content}>
+            <CardAboutPhrase
+              title={strings.recoveryPhraseCardTitle}
+              items={[
+                strings.recoveryPhraseCardFirstItem,
+                strings.recoveryPhraseCardSecondItem,
+                strings.recoveryPhraseCardThirdItem,
+                strings.recoveryPhraseCardFourthItem,
+                strings.recoveryPhraseCardFifthItem,
+              ]}
+            />
+
+            <LearnMoreButton
+              onPress={() => {
+                ;('')
+              }}
+            />
+          </View>
+        </ScrollView>
+
+        <Space height="s" />
+
+        <Button title={strings.continueButton} style={styles.button} onPress={closeModal} />
+
+        <Space height="l" />
+      </View>,
+      HEIGHT_MODAL,
+    )
+  }
+
+  return (
+    <SafeAreaView edges={['left', 'right', 'bottom']} style={styles.container}>
+      <StatusBar type="light" />
+
+      <View style={styles.content}>
+        <StepperProgress currentStep={2} currentStepTitle={strings.recoveryStepper} totalSteps={4} displayStepNumber />
+
+        <Text style={styles.title}>
+          {strings.recoveryPhraseTitle}
+
+          <Pressable onPress={showModal}>
+            <InfoIcon />
+          </Pressable>
+        </Text>
+
+        <View style={styles.mnemonicWords}>
+          <BlurView intensity={isBlur ? 14 : 0} style={styles.blurView} />
+
+          {mockAddWallet.mnemonic.split(' ').map((word, index) => (
```

Comment:
it should receive through navigation or context. UI logic + navigation is part of SoW